ipn/ipnserver/actor.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package ipnserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"tailscale.com/ipn"
+	"tailscale.com/ipn/ipnauth"
+	"tailscale.com/types/logger"
+	"tailscale.com/util/osuser"
+	"tailscale.com/version"
+)
+
+var _ ipnauth.Actor = (*actor)(nil)
+
+// actor implements [ipnauth.Actor] and provides additional functionality that is
+// specific to the current (as of 2024-08-27) permission model.
+//
+// Deprecated: this type exists for compatibility reasons and will be removed as
+// we progress on tailscale/corp#18342.
+type actor struct {
+	logf logger.Logf
+	ci   *ipnauth.ConnIdentity
+
+	isLocalSystem bool // whether the actor is the Windows' Local System identity.
+}
+
+func newActor(logf logger.Logf, c net.Conn) (*actor, error) {
+	ci, err := ipnauth.GetConnIdentity(logf, c)
+	if err != nil {
+		return nil, err
+	}
+	return &actor{logf: logf, ci: ci, isLocalSystem: connIsLocalSystem(ci)}, nil
+}
+
+// IsLocalSystem implements [ipnauth.Actor].
+func (a *actor) IsLocalSystem() bool {
+	return a.isLocalSystem
+}
+
+// IsLocalAdmin implements [ipnauth.Actor].
+func (a *actor) IsLocalAdmin(operatorUID string) bool {
+	return a.isLocalSystem || connIsLocalAdmin(a.logf, a.ci, operatorUID)
+}
+
+// UserID implements [ipnauth.Actor].
+func (a *actor) UserID() ipn.WindowsUserID {
+	return a.ci.WindowsUserID()
+}
+
+func (a *actor) Pid() int {
+	return a.ci.Pid()
+}
+
+// Username implements [ipnauth.Actor].
+func (a *actor) Username() (string, error) {
```

Comment:
And this is https://github.com/tailscale/tailscale/blob/b78df4d48ab81e211650d33fcbdaecb7b6f6b87d/ipn/localapi/localapi.go#L1139-L1165